### ISSUE TITLE
ride breakdown leads to block brake failure leads to ride crash

### ISSUE THREAD

**Issue by cheweytoo** [OpenRCT2/OpenRCT2#9893]
**OS:** Linux x86_64
**Version:**  v0.2.3-dev
**Commit/Build:** 406ec7f
Build downloaded from openrct2.org

TL;DR: Just let attached savegame run for two minutes and watch the ride crash.

When the ride in the attached savegame breaks down with "Restraints stuck closed" or "Restraints stuck open", the second block brake fails to stop the train, leading immediately to more than one train per section, and a lethal crash after the ride restarts post-repair.

I did my best to build a reduced testcase, and **in the attached park**, this happens reliably when a ride breakdown is triggered. However:

- I have _not_ immediately been able to reproduce this with another ride layout of the same ride type, or another ride type altogether.
- It seems to be necessary to have people on the ride.

- [ ] Reproducible in RCT2 (vanilla)?
No idea, I only run Linux and OpenRCT2
- [ ] Specific to multiplayer?
No.

**Steps to reproduce:**
I'm not sure about generic STR, but see above, and attached savegame.

**Screenshots / Video:**
![crash_with_block_brakes](https://user-images.githubusercontent.com/29800630/63214946-86ba1600-c11f-11e9-9887-0430ce318ecd.png)

**Save game:**
[crash_with_block_brakes.txt](https://github.com/OpenRCT2/OpenRCT2/files/3512213/crash_with_block_brakes.txt)

**Comment by jensj12** [comment]
The ride doesn't need to break down, set the minimum wait time to 30 sec and it will crash as well. Looks like the block brakes simply aren't strong enough to bring a full train to a complete stop.

**Comment by cheweytoo** [comment]
Ohhh, interesting. I wasn't aware that block brakes have an entry speed limit; makes sense though.
I managed to hit the rare corner case of "an empty train is stopped, a full one isn't" apparently, which explains my troubles with building a reduced testcase.

So this is simply intentional and not a bug, right? Just making sure, because your "for now" could be understood as "…until the bug is fixed".

**Comment by duncanspumpkin** [comment]
Yes this is intentional. Block brakes have a max speed

**Comment by cheweytoo** [comment]
Got it. Thanks!
